sklearn/preprocessing/_label.py

```diff
@@ -303,7 +304,8 @@ def fit(self, y):
             raise ValueError("y has 0 samples: %r" % y)
 
         self.sparse_input_ = sp.issparse(y)
-        self.classes_ = unique_labels(y)
+        xp, _ = get_namespace(y)
+        self.classes_ = _convert_to_numpy(unique_labels(y), xp)
```

Comment:
I think that so far the policy is: store the fitted attributes with the array type that makes most sense to be efficient at prediction time assuming the prediction-time data container type will be consistent with the fit-time data container type. Since this PR only changes `LabelBinarizer` for convenience without actually delegating any computation to the underlying Array API namespace (see the inline comments), I think it's better to always keep `classes_` as a numpy array for now.

If one day we decide to recode `label_binarize` to actually delegate some computation to the underlying namespace, then we think about make the type of `classes_` input dependent instead.

Reply:
> Since this PR only changes LabelBinarizer for convenience without actually delegating any computation to the underlying Array API namespace (see the inline comments), I think it's better to always keep classes_ as a numpy array for now.

Then I'm confused. Doesn't `check_array` already convert data to `numpy`? So we already _support_ non-numpy data.

If the point is to have the output of `predict` in the same space as what user gives, shouldn't that be like a decorator or something around `predict`? Or a simply function call at the end of predict?

Reply:
> Or a simply function call at the end of predict?

that is quite close to what is done here: we record the input namespace and device at the beginning of the function, convert to numpy and do everything in numpy, and convert back to the input format and device where the function returns.

As you say, it could probably be implemented as a decorator applied to `fit`, `label_binarize` and `inverse_transform`